UefiCpuPkg/Library/MpInitLib/MpLib.c

```diff
@@ -2249,13 +2250,22 @@ MpInitLibInitialize (
   // EfiBootServicesCode to avoid page fault if NX memory protection is enabled.
   //
   CpuMpData->WakeupBufferHigh = AllocateCodeBuffer (ApResetVectorSizeAbove1Mb);
+
+  Cr0.UintN = AsmReadCr0 ();
+  if (Cr0.Bits.PG != 0) {
+    RemoveNxProtection ((EFI_PHYSICAL_ADDRESS)(UINTN)CpuMpData->WakeupBufferHigh, ALIGN_VALUE (ApResetVectorSizeAbove1Mb, EFI_PAGE_SIZE));
+  }

```

Comment:
Can you rename AllocateCodeBuffer() to AllocateCodePage()?
So that we guarantee that RemoveNxProtection() does not remove NX for memory region after "AP Reset Vector Above 1MB".

Reply:
Done in latest.